scripts/generate-icons.ts

```diff
@@ -0,0 +1,343 @@
+						resolve();
+					});
+				} catch (err) {
+					reject(err);
+				}
+			});
+		},
+	};
+}
+
+/**
+ * Creates a png from a svg without making any edits to it.
+ *
+ * @param folder - folder containing the icon svg in src.
+ * @param path - name of the svg file.
+ * @param size - desired resolution of the png file.
+ * @returns a PNG buffer from the svg file
+ */
+async function createUnmodifiedIcon(
+	folder: string,
+	path: string,
+	size: number
+) {
+	const canvas = createCanvas(size, size);
+	const ctx = canvas.getContext('2d');
+	const image = await loadImage(resolve(input, folder, path));
+	ctx.drawImage(image, 0, 0, size, size);
+	return canvas.toBuffer();
+}
+
+/**
+ * Creates a png from a svg, making it into a monochrome icon of a specified color
+ *
+ * @param path - name of the svg file.
+ * @param size - desired resolution of the png file.
+ * @param color - color code of the color to be used.
+ * @returns a monochrome PNG buffer from the svg file.
+ */
+async function createMonochromeIcon(
+	path: string,
+	size: number,
+	color: string,
+	borderColor?: string
+) {
+	// If there is a border, make room for the border
+	const iconSize = borderColor ? size - borderThickness * 2 : size;
+
+	// draw the icon itself to one canvas first
+	const iconCanvas = createCanvas(size, size);
+	const iconCtx = iconCanvas.getContext('2d');
+	const image = await loadImage(resolve(input, 'monochrome', path));
+	iconCtx.drawImage(
+		image,
+		borderThickness,
+		borderThickness,
+		iconSize,
+		iconSize
+	);
+	// fill it in with the main icon color
+	iconCtx.globalCompositeOperation = 'source-in';
+	iconCtx.fillStyle = color;
+	iconCtx.fillRect(0, 0, size, size);
+
+	// In safari we don't need borders ever. Skip the next part and return immediately
+	if (!borderColor) {
+		return iconCanvas.toBuffer();
+	}
+
+	// now draw icon border
+	const canvas = createCanvas(size, size);
+	const ctx = canvas.getContext('2d');
+	ctx.drawImage(image, 0, 0, size, size);
+	// fill in
+	ctx.globalCompositeOperation = 'source-in';
+	ctx.fillStyle = borderColor;
+	ctx.fillRect(0, 0, size, size);
+
+	// draw the main icon inside the border
+	ctx.globalCompositeOperation = 'source-over';
+	ctx.drawImage(iconCanvas, 0, 0, size, size);
+
+	return canvas.toBuffer();
+}
+
+/**
+ * Gets new PNG file name for page action icons
+ *
+ * @param path - original svg filename.
+ * @param size - desired resolution of png.
+ * @param type - type of icon (e.g. light/dark/safari)
+ * @returns PNG file name to output to.
+ */
+function getOutputPath(path: string, size: number, type: string) {
+	const name = path.split('.')[0];
+	return resolve(output, `${name}_${size}_${type}.png`);
+}
+
+/**
+ * Writes all the appropriate PNG files for a single monochrome svg.
+ *
+ * @param path - filename of svg.
+ */
+async function writeMonochromeIcon(
+	path: keyof typeof monochromeColors
+): Promise<void> {
+	if (releaseTarget === 'safari') {
```

Comment:
We could have an enum/type that represents these release targets - but could be a future iteration :)

Reply:
Did it anyway